### PR TITLE
Update Service Bus SDK from t1 to t2

### PR DESCRIPTION
**Pull request by v-yilinhu**
1. Update Service Bus SDK from t1 to t2

2. Update files:

    - ./ThroughputTest/Program.cs
    - ./ThroughputTest/Properties/launchSettings.json
    - ./ThroughputTest/ReceiverTask.cs
    - ./ThroughputTest/SenderTask.cs
    - ./ThroughputTest/Settings.cs
    - ./ThroughputTest/ThroughputTest.csproj

@scottaddie, @AlexGhiondea, @lilyjma, @ramya-rao-a and @JoshLove-msft for notification.

### CODE REVIEW

**Comment by ramya-rao-a** [comment]
@JoshLove-msft, please review

File: ThroughputTest/SenderTask.cs [hunk 1]
```diff
@@ -148,27 +148,29 @@ private async Task HandleExceptions(DynamicSemaphoreSlim semaphore, SendMetrics
             bool wait = false;
             ex.Handle((x) =>
             {
-                if (x is ServiceBusCommunicationException)
+                if ((ServiceBusException)x is ServiceBusException)
```
Comment (JoshLove-msft) [inline ThroughputTest/SenderTask.cs:151]:
```suggestion
                if (x is ServiceBusException sbException)
```

File: ThroughputTest/SenderTask.cs [hunk 2]
```diff
@@ -148,27 +148,29 @@ private async Task HandleExceptions(DynamicSemaphoreSlim semaphore, SendMetrics
...
                 {
-                    if (((ServiceBusCommunicationException)x).InnerException is SocketException &&
-                        ((SocketException)((ServiceBusCommunicationException)x).InnerException).SocketErrorCode == SocketError.HostNotFound)
+                    if (((ServiceBusException)x).Reason == ServiceBusFailureReason.MessagingEntityNotFound)
```
Comment (JoshLove-msft) [inline ThroughputTest/SenderTask.cs:153]:
```suggestion
```

File: ThroughputTest/SenderTask.cs [hunk 3]
```diff
@@ -148,27 +148,29 @@ private async Task HandleExceptions(DynamicSemaphoreSlim semaphore, SendMetrics
...
                     {
-                        return false;
+                        if (((ServiceBusException)x).InnerException is SocketException &&
+                        ((SocketException)((ServiceBusException)x).InnerException).SocketErrorCode == SocketError.HostNotFound)
```
Comment (JoshLove-msft) [inline ThroughputTest/SenderTask.cs:156]:
```suggestion
                        if (sbException.InnerException is SocketException socketException &&
                        socketException.SocketErrorCode == SocketError.HostNotFound)
```

File: ThroughputTest/SenderTask.cs [hunk 4]
```diff
@@ -148,27 +148,29 @@ private async Task HandleExceptions(DynamicSemaphoreSlim semaphore, SendMetrics
...
+                            return false;
+                        }
                     }
-                }
-
-                if (x is ServerBusyException)
-                {
-                    sendMetrics.BusyErrors = 1;
-                    if (!this.CancellationToken.IsCancellationRequested)
+                    if (((ServiceBusException)x).Reason == ServiceBusFailureReason.ServiceBusy)
```
Comment (JoshLove-msft) [inline ThroughputTest/SenderTask.cs:161]:
```suggestion
                    if (sbException.Reason == ServiceBusFailureReason.ServiceBusy)
```

File: ThroughputTest/ThroughputTest.csproj [hunk 5]
```diff
@@ -7,9 +7,9 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Azure.Messaging.ServiceBus" Version="7.4.0" />
```
Comment (JoshLove-msft) [inline ThroughputTest/ThroughputTest.csproj:10]:
```suggestion
    <PackageReference Include="Azure.Messaging.ServiceBus" Version="7.5.0" />
```

File: ThroughputTest/ReceiverTask.cs [hunk 6]
```diff
@@ -139,84 +141,47 @@ async Task ReceiveTask(string path)
...
-                                    // batch complete
-                                    await receiver.CompleteAsync(t.Result.Select((m) => { return m.SystemProperties.LockToken; })).ContinueWith(async (t1) =>
+                                    await Task.Delay(TimeSpan.FromMilliseconds(Settings.WorkDuration));
+                                    await receiver.CompleteMessageAsync(t.Result[i]).ContinueWith(async (t1) =>
                                     {
                                         receiveMetrics.CompleteDuration100ns = sw.ElapsedTicks - nsec;
                                         if (t1.IsFaulted)
                                         {
-                                            if (t1.Exception?.GetType() == typeof(ServerBusyException))
+                                            if (t1.Exception?.GetType() == typeof(ServiceBusFailureReason))
```
Comment (JoshLove-msft) [inline ThroughputTest/ReceiverTask.cs:184]:
```suggestion
                                            if (t1.Exception is ServiceBusException sbException && sbException.Reason == ServiceBusFailureReason.ServiceBusy)
```

File: ThroughputTest/ReceiverTask.cs [hunk 7]
```diff
@@ -232,37 +197,71 @@ async Task ReceiveTask(string path)
                                         else
                                         {
                                             receiveMetrics.CompleteCalls = 1;
-                                            receiveMetrics.Completions = t.Result.Count;
+                                            receiveMetrics.Completions = 1;
```
Comment (JoshLove-msft) [inline ThroughputTest/ReceiverTask.cs:200]: Why was this changed?

File: ThroughputTest/ReceiverTask.cs [hunk 8]
```diff
@@ -139,84 +141,47 @@ async Task ReceiveTask(string path)
                 {
                     nsec = sw.ElapsedTicks;
                     // we're going to unblock the receives after 10 seconds if there's no pending message
-                    receiver.ReceiveAsync(Settings.ReceiveBatchCount, TimeSpan.FromSeconds(10)).ContinueWith(async (t) =>
+                    Task<IReadOnlyList<ServiceBusReceivedMessage>> t = receiver.ReceiveMessagesAsync(Settings.ReceiveBatchCount, TimeSpan.FromSeconds(10));
```
Comment (JoshLove-msft) [inline ThroughputTest/ReceiverTask.cs:144]: Why was the ContinueWith removed?

Comment (v-yilinhu) [reply ThroughputTest/ReceiverTask.cs]: @JoshLove-msft I have updated, please review.

File: ThroughputTest/ReceiverTask.cs [hunk 9]
```diff
@@ -47,8 +46,11 @@ protected override Task OnStartAsync()
 
         async Task ReceiveTask(string path)
         {
-            var receiver = new MessageReceiver(this.Settings.ConnectionString, path, this.Settings.ReceiveMode);
-            receiver.PrefetchCount = Settings.PrefetchCount;
+            ServiceBusClient client = new ServiceBusClient(this.Settings.ConnectionString);
+            ServiceBusReceiverOptions options = new ServiceBusReceiverOptions();
```
Comment (scottaddie) [inline ThroughputTest/ReceiverTask.cs:50]:
```suggestion
            var client = new ServiceBusClient(this.Settings.ConnectionString);
            var options = new ServiceBusReceiverOptions();
```

File: ThroughputTest/SenderTask.cs [hunk 10]
```diff
@@ -95,12 +95,12 @@ async Task SendTask()
                 }
                 else
                 {
-                    List<Message> batch = new List<Message>();
+                    List<ServiceBusMessage> batch = new List<ServiceBusMessage>();
```
Comment (scottaddie) [inline ThroughputTest/SenderTask.cs:98]:
```suggestion
                    var batch = new List<ServiceBusMessage>();
```

**Review by scottaddie — APPROVED** [review]

**Comment by v-xuto** [comment]
@jongio Please help merge this PR.
